File: 4/jessica_ng/picmaker.py
import random

f = open ("pic.ppm", 'w')
f.write ("P3\n500 500\n255\n")

i = 500 * 500
n = random.randint (0, 255)
x = n%3

while i > 0:
    if x == 0:
        r = i%255
        g = n
        b = n
    elif x == 1:
        r = n
        g = i%255
        b = n
    else:
        r = n
        g = n
        b = i%255
        
    t = ("%d %d %d "%(r, g, b))
    i = i - 1
    f.write (t)

f.close()

File: 4/Dmitriy_Kagno/picmaker.c
#include <unistd.h>
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <fcntl.h>
#include <sys/stat.h>
#include <sys/types.h>


int sq(int x){

  return x * x;
}

int main(){

  int red, blue, green;
  int fd = open("pic.ppm", O_CREAT | O_TRUNC | O_WRONLY, 0644);

  char line[256];

  sprintf( line, "P3\n500 500\n255\n" );
  write( fd, line, strlen(line) );


  int h,k;
  h = 250;
  k = 250;

  int i,j;
  for (i = 0; i<500; i++){
    for (j = 0; j < 500; j++){

      red = 0;
      green = 0;
      blue = 0;
      
      if((sq(i-h) + (sq(j-k))) == sq(100) ){
	red = 255;
	green = 255;
	blue = 255;
      }

      
      /*if((i+j)%2){
	red = i%255;
	green = j%255;
	blue = 255;
      }   
      */

      sprintf( line, "%d %d %d ", red, green, blue);
      write( fd, line, strlen(line) );
  
    }
  }




  close(fd);
  
}
